pkg/storage/storage.go

```diff
@@ -168,10 +170,16 @@ type RelationshipTupleWriter interface {
 // ReadStartingWithUserFilter specifies the filter options that will be used
 // to constrain the [RelationshipTupleReader.ReadStartingWithUser] query.
 type ReadStartingWithUserFilter struct {
+	// Mandatory.
 	ObjectType string
-	Relation   string
+	// Mandatory.
+	Relation string
+	// Mandatory.
 	UserFilter []*openfgav1.ObjectRelation
-	ObjectIDs  []string // Optional. If present, the datastore returns the intersection between this filter and what is in the database
+
+	// Optional. If present, it will be sorted in ascending order.
```

Comment:
consider whether we really need this to be ascending order OR have the individual datastore implementation re-order as needed

Reply:
If we require all datastore implementations to sort then I might as well scrap this PR 🤔

Reply:
In this case, should we rename it to SortedObjectIDs to make it very clear that it is to be sorted.  (Though this will be a breaking change with respect to the interface)

Reply:
To avoid all this, why not just pass the sorted set directly?